cmd/server/main.go

```diff
@@ -372,6 +373,24 @@ func run(args []string) error {
 	return nil
 }
 
+var publicEndpoints = []*regexp.Regexp{
+	regexp.MustCompile(`^/initiative/[^/]*$`),
```

Comment:
Ignoring my disdain for regexes for a second, I think we might want something a bit more flexible here, something like:

```go
type allowFn func(r *http.Request) bool
var publicEndpoints = []allowFn{
  allowInitiatives,
  // other stuff in the future
}

func allowInitiatives(r *http.Request) bool {
  return yourRegexp.MatchString(r.URL.path) && r.Method == http.MethodGet
}
```

The reason I suggest this is because I don't think you actually want "all paths starting with `"/initiative/"` and at most one sub-path", as that includes `POST` requests and stuff, so you'll want to check `r.Method` as well.

That said, I do also think the regex is overkill here, I'd explicitly enumerate paths instead, as that's a safer invariant as our API surface changes

Reply:
Really good idea - though the reason we (probably) need regex as opposed to enumerating paths is that the second path component is the ID: `/initative/some-id-here`, and since that can be anything and be valid, we can't enumerate paths.

Reply:
So TL;DR: I'll use the `allowFn` syntax, and also exclude non-get requests, but keep that particular regex.